pyartcd/pyartcd/pipelines/ocp4_scan_konflux.py

```diff
@@ -125,15 +132,47 @@ async def ocp4_scan(runtime: Runtime, version: str, assembly: str, data_path: st
                     image_list: str, ignore_locks: bool):
     jenkins.init_jenkins()
 
-    # TODO handle locks
-    if ignore_locks and not runtime.dry_run:
-        raise RuntimeError('--ignore-locks can only by used with --dry-run')
+    lock = Lock.SCAN_KONFLUX
+    lock_name = lock.value.format(version=version)
+    lock_identifier = jenkins.get_build_path()
+    if not lock_identifier:
+        runtime.logger.warning('Env var BUILD_URL has not been defined: a random identifier will be used for the locks')
 
-    await Ocp4ScanPipeline(
+    pipeline = Ocp4ScanPipeline(
         runtime=runtime,
         version=version,
         assembly=assembly,
         data_path=data_path,
         data_gitref=data_gitref,
         image_list=image_list,
-    ).run()
+    )
+
+    if ignore_locks:
+        if not runtime.dry_run:
+            raise RuntimeError('--ignore-locks can only by used with --dry-run')
+        await pipeline.run()
+
+    else:
+        async def run_with_build_lock():
+            build_lock = Lock.BUILD_KONFLUX
+            build_lock_name = build_lock.value.format(version=version)
+            await locks.run_with_lock(
+                coro=pipeline.run(),
+                lock=build_lock,
+                lock_name=build_lock_name,
+                lock_id=lock_identifier,
+                skip_if_locked=True
+            )
+
+        await locks.run_with_lock(
+            coro=run_with_build_lock(),
+            lock=lock,
+            lock_name=lock_name,
+            lock_id=lock_identifier,
+            skip_if_locked=True
+        )
+
+    # This should not happen, as the schedule build already checked for existing locks
+    # It doesn't hurt to check once more though
```

Comment:
I don't know if this comment is relevant here.. since the below statement will be checked for each pipeline run